### ISSUE TITLE
"Entry address not saved on coroutine"

### ISSUE THREAD

**Issue by longouyang** [probmods/ppaml2016#4]
There's an example in the variational subchapter:

``` js
gaussian(0, 1, {
    guide: Gaussian({mu: scalarParam(0, 1), sigma: Math.exp(scalarParam(0, 1))})
});
```

that causes the error:

```
Entry address not saved on coroutine
```

(At time of writing, we're using webppl 0.8.2 from the master branch)

**Comment by null-a** [comment]
Thanks @longouyang. The problem comes from using `scalarParam` outside of a coroutine, I've improved this in probmods/webppl#588.

**Comment by null-a** [comment]
Closed by probmods/webppl#588.
